File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& str) {
        vector<string> strs = str;

        for (int i = 0; i < strs.size(); i++) {
            sort(strs[i].begin(), strs[i].end());
        }

        unordered_map<string, vector<string>> mp;

        for (int i = 0; i < strs.size(); i++) {
            mp[strs[i]].push_back(str[i]);
        }

        vector<vector<string>> g;

        for (auto &it : mp) {
            g.push_back(it.second);
        }

        return g;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        sorted_str=strs[:]

        for i in range(len(strs)):
            sorted_str[i]=''.join(sorted(sorted_str[i]))

        mp={}

        for i in range(len(sorted_str)):
            if sorted_str[i] not in mp:
                mp[sorted_str[i]]=[]
            mp[sorted_str[i]].append(strs[i])
        

        ans=[]

        for key in mp:
            ans.append(mp[key])

        return ans

        